openedx/core/djangoapps/user_api/accounts/tests/test_api.py

```diff
@@ -191,6 +192,25 @@ def test_serializer_save_fails(self, serializer_save):
         pending_change = PendingEmailChange.objects.filter(user=self.user)
         self.assertEqual(0, len(pending_change))
 
+    def test_language_proficiency_eventing(self):
+        """
+        Test that eventing of language proficiencies, which happens update_account_settings method, behaves correctly.
+        """
+        def verify_event_emitted(new_value, old_value):
+            update_account_settings(self.user, {"language_proficiencies": new_value})
+            self.assert_event_emitted(
+                USER_SETTINGS_CHANGED_EVENT_NAME, setting="language_proficiencies",
+                old=old_value, new=new_value, user_id=self.user.id, table="student_languageproficiency"
+            )
+            self.reset_tracker()
+
+        # Change language_proficiencies and verify events are fired.
+        verify_event_emitted([{"code": "en"}], [])
```

Comment:
Gurrh... this test is still failing (since Andy switched the code to call the helper method). There's something I'm not seeing, but it will have to wait until morning.